campaignion_tracking/js/campaignion_tracking.js

```diff
@@ -0,0 +1,82 @@
+(function($) {
+Drupal.behaviors.campaignion_tracking = {};
+Drupal.behaviors.campaignion_tracking.attach = function(context, settings) {
+
+  /**
+   * Dispatch to campaignion_tracking.tracker.
+   *
+   * Nothing happens if the tracker is not available.
+   */
+
+  function gracefulDispatch (name, data, trackingContext) {
+    // Build an msg object expected by campaignion_tracking:
+    // It has a "name" for the event, "data" for the event and a
+    // "context" for additional data.
+    // NB: the event handler of the tracker might expect a certain data
+    // structure.
+    var msg = {
+      'name': name,
+      'data': data,
+      'context': trackingContext
+    }
+    window.campaignion_tracking.tracker.publish('donation', msg)
+  }
+
+  /**
+   * Donation related events.
+   *
+   * The events are only fired if a correct campaignion_tracking context is
+   * available. They are only fired once for the Drupal context to prevent the
+   * same messages being dispatch multiple times (behaviours can be called
+   * multiple times).
+   *
+   * Events:
+   * - 'setDonationProduct'
+   * - 'checkoutBegin'
+   * - 'checkoutEnd'
+   */
+  if (settings['campaignion_tracking'] && settings['campaignion_tracking']['context']) {
+    settings['campaignion_tracking']['sent'] = settings['campaignion_tracking']['sent'] || [];
+
+    var node = settings.campaignion_tracking.context['node'] || {};
+    var donation = settings.campaignion_tracking.context['donation'] || {};
+    var webform = settings.campaignion_tracking.context['webform'] || {};
+
+    if (node['is_donation']) {
```

Comment:
Would it be better to dispatch the events during the submit-events on the form? — Maybe something to be discussed broken out into another ticket.

Reply:
yes, agree, would be better. also yes, maybe best in a new card...